### ISSUE TITLE
Node esm support

### ISSUE THREAD

**Issue by daKmoR** [lucaong/minisearch#152]
While I was testing to add an export map I found the following problem.

e.g. if we add this to minisearch

```json
  "exports": {
    ".": {
      "types": "./dist/types/index.d.ts",
      "require": "./dist/umd/index.js",
      "default": "./dist/es/index.js"
    }
  },
```

and then try to use it in node

```
// package.json
  "type": "module"
// test.js
import MiniSearch from 'minisearch';

// execute
node test.js
```

we get the following error 

```
node test.js
(node:65561) Warning: To load an ES module, set "type": "module" in the package.json or use the .mjs extension.
(Use `node --trace-warnings ...` to show where the warning was created)
/www/testmini/node_modules/minisearch/dist/es/index.js:1436
export default MiniSearch;
^^^^^^

SyntaxError: Unexpected token 'export'
```

which is correct as we try to import the file `dist/es/index.js` but the package `minisearch` is define as a `require` package... e.g. it has an implicit `"type": "commonjs",`. e.g. all `*.js` files are treated as common js

## Solutions

1. add `type: "module"` to package.json to minisearch so that `*.js` files are treated as modules (and export the commonjs version as *.cjs files) 
2. export the module version as `*.mjs` (which is kinda strange for a browser but still works)

What do you think?

**Comment by lucaong** [comment]
Hi @daKmoR ,
thanks for the summary. I have to admit I am not yet well versed with the latest Node settings for modules. In general, since most users of MiniSearch use it in the browser, I would privilege a solution that is the least surprising for browser users, both those using the `npm` package and those using a CDN. Option 1 therefore seems therefore the most appropriate.

Would there be repercussions on which Node versions are supported? I am fine with breaking compatibility with very old versions if it's hard to support them all, but I would like to assess the situation.

**Comment by daKmoR** [comment]
you can probably support "all" the node versions if you want

something like this

```
"type": "module",
// node <12
"main": "./dist/cjs/index.cjs"
// node 12+
"exports": {
   // ts 4.7+
   "types": "./dist/types/index.d.ts",
   "require": "./dist/cjs/index.cjs",
   "default": "./dist/esm/index.js"
}
// ts < 4.7
"types": "./dist/types/index.d.ts",
```

and then I guess you would also have a separate umd for CDNs?

**Comment by lucaong** [comment]
This sounds good, and could be tested in a `5.0.0-beta2` release on NPM to be sure.

**Comment by lucaong** [comment]
> and then I guess you would also have a separate umd for CDNs?

Or the `main` could be the UMD version, what do you think @daKmoR ?

My line of thought: if an environment doesn’t know about ES modules, etc. better to just give the version that is compatible with the most approaches (UMD), as it could be an old version of Node, or a build system for a browser, or even a CDN used with a `script` tag, while if the environment understands `exports`, let it choose the most appropriate.

**Comment by daKmoR** [comment]
sounds good to me 👍

**Comment by lucaong** [comment]
@daKmoR I implemented this in #158

**Comment by lucaong** [comment]
Released on NPM as part of `v5.0.0-beta2`
